src/modules/flow/process/subprocess.c

```diff
@@ -172,9 +172,9 @@ on_fork(void *data)
  */
 err:
     SOL_WRN("Failed in setup the files descriptors");
-    close(mdata->pipes.stdout[1]);
-    close(mdata->pipes.stderr[0]);
-    close(mdata->pipes.stdin[0]);
+    close(mdata->pipes.out[1]);
```

Comment:
Also unrelated to the patch, but maybe `0` and `1` could be changed to constants, such as `READ_SIDE` or `WRITE_SIDE`?